apps/web/scripts/kernel/handlers/solanaHandler.ts

```diff
@@ -1,58 +1,92 @@
 import { Keypair, VersionedTransaction } from '@solana/web3.js';
-import { MessengerCallback } from '@walless/messaging';
+import { Networks } from '@walless/core';
+import { MessengerCallback, ResponseCode } from '@walless/messaging';
 import { signAndSendTransaction, signMessage } from '@walless/network';
 import { decode, encode } from 'bs58';
 
 import { solanaConnection } from '../utils/connection';
 import { triggerActionToGetPrivateKey } from '../utils/handler';
 
+import { getPrivateKey, settings } from './../utils/handler';
+
 export const handleSignTransaction: MessengerCallback = async (
 	payload,
 	channel,
 ) => {
 	const privateKey = await triggerActionToGetPrivateKey();
+
 	if (!privateKey) {
 		return;
 	}
+
 	const serializedTransaction = decode(payload.transaction);
 	const transaction = VersionedTransaction.deserialize(serializedTransaction);
 	const keypair = Keypair.fromSecretKey(privateKey);
+
 	transaction.sign([keypair]);
+
 	channel.postMessage({
 		from: 'walless@kernel',
 		requestId: payload.requestId,
 		signedTransaction: encode(transaction.serialize()),
 	});
+
 	return transaction.serialize();
 };
 
 export const handleSignAndSendTransaction: MessengerCallback = async (
 	payload,
 	channel,
 ) => {
+	const messageHeader = {
+		from: 'walless@kernel',
+		requestId: payload.requestId,
+	};
+
+	if (settings.requirePasscode && !payload.passcode) {
+		return channel.postMessage({
+			...messageHeader,
+			responseCode: ResponseCode.REQUIRE_PASSCODE,
+		});
+	}
+
 	// Prepare private key
-	const privateKey = await triggerActionToGetPrivateKey();
-	if (!privateKey) {
-		return;
+	let privateKey;
+	try {
+		privateKey = await getPrivateKey(Networks.solana, payload.passcode);
+	} catch (error) {
+		return channel.postMessage({
+			...messageHeader,
+			responseCode: ResponseCode.WRONG_PASSCODE,
+			message: (error as Error).message,
+		});
 	}
 
 	// Transaction object
 	const serializedTransaction = decode(payload.transaction);
 	const transaction = VersionedTransaction.deserialize(serializedTransaction);
 
-	const signatureString = await signAndSendTransaction(
-		solanaConnection,
-		transaction,
-		payload.options || {},
-		privateKey,
-	);
+	let signatureString;
+	try {
+		signatureString = await signAndSendTransaction(
+			solanaConnection,
+			transaction,
+			payload.options || {},
+			privateKey as Uint8Array,
+		);
```

Comment:
if possible: with 4 params or more, that will be useful to use an object as params instead.

Reply:
We may refactor all of them after completing tempory methods (with temp passcode)